airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/up_to_date/pipeline.py

```diff
@@ -0,0 +1,326 @@
+
+    async def _run(self) -> StepResult:
+        connector_dir_entries = await (await self.context.get_connector_dir()).entries()
+        if self.context.connector.language not in [ConnectorLanguage.PYTHON, ConnectorLanguage.LOW_CODE]:
+            return StepResult(
+                step=self,
+                status=StepStatus.SKIPPED,
+                stderr="The connector is not a Python connector.",
+            )
+        if self.context.connector.connector_type != "source":
+            return StepResult(
+                step=self,
+                status=StepStatus.SKIPPED,
+                stderr="The connector is not a source connector.",
+            )
+        if not "poetry.lock" in connector_dir_entries or not "pyproject.toml" in connector_dir_entries:
+            return StepResult(
+                step=self,
+                status=StepStatus.SKIPPED,
+                stderr="The connector requires poetry.",
+            )
+
+        if not self.context.connector.metadata or not self.context.connector.metadata.get("connectorBuildOptions", {}).get("baseImage"):
+            return StepResult(
+                step=self,
+                status=StepStatus.SKIPPED,
+                stderr="The connector CDK can't be updated because it does not have a base image defined in the metadata.",
+            )
+
+        # TODO: is there a fast way to check if the connector is already using the latest CDK version?
+        # this is better than all the docker shenanigans if it's not needed.
+        # probably just grep the toml file. If can't find it fail over to the better way
+
+        return StepResult(
+            step=self,
+            status=StepStatus.SUCCESS,
+        )
+
+
+class UpdateCdk(Step):
+    context: ConnectorContext
+
+    title = "Update the CDK to a new version."
+
+    def __init__(self, context: PipelineContext, cdk_version: str) -> None:
+        super().__init__(context)
+        self.cdk_version = cdk_version
+
+    async def _run(self) -> StepResult:
+        base_image_name = self.context.connector.metadata["connectorBuildOptions"]["baseImage"]
+        base_container = self.dagger_client.container(platform=LOCAL_BUILD_PLATFORM).from_(base_image_name)
+        connector_container = await with_python_connector_installed(
+            self.context,
+            base_container,
+            str(self.context.connector.code_directory),
+        )
+
+        try:
+            poetry_show_result = await connector_container.with_exec(sh_dash_c(["poetry show | grep airbyte-cdk"])).stdout()
+            if not poetry_show_result:
+                return StepResult(step=self, status=StepStatus.FAILURE, stderr="CDK is not installed in the connector.")
+            current_cdk_version = extract_poetry_show_version(poetry_show_result)
+            if not current_cdk_version:
+                return StepResult(step=self, status=StepStatus.FAILURE, stderr="CDK was not found in the connector poetry.")
```

Comment:
This feels like it belongs in `CheckIsUpdateCdkCandidate`. Any reason not to put it there?

Reply:
This was back to when I spun up the container. I tried to do it all in one spot. Is that common?

Reply:
Got it, that makes sense to me.

Reply:
In the world we are moving to that might do one of 7 things, should there be 7 different checks like this? Or should there be 7 tasks that might return `SKIPPED` - i lean towards the later because otherwise, you end up with duplicated logic between the Candidate step and the ActuallyDoIt step.